Resolve #107 Show Shapefile Download Icon

mzagaja opened the pull request:
Show shapefile download button for the appropriate kinds of datasets. Now we will stop showing them for tabular datasets and show it for geospatial datasets.

mzagaja committed:
Resolve #107 Show Shapefile Download Icon

Show shapefile download button for the appropriate kinds of datasets. Now we will stop showing them for tabular datasets and show it for geospatial datasets.

mzagaja committed:
Update Shapefile Download Link

This commit resolves MAPC/datacommon#107 by updating the endpoint for shapefiles to point to the new shapefile endpoint. It also removes the GeoJSON endpoint because it no longer works without Carto.

ericyoungberg commented on app/templates/datasets.hbs:
```diff
@@ -54,14 +54,11 @@
               </a>
-              <a class="button lift" href={{download_link_geojson}} download={{model.dataset.table_name}}>
-                .geojson
-              </a>
```
This PR removes GeoJSON support?

mzagaja replied:
Not specifically but GeoJSON is dependent upon CartoDB so we would need to build GeoJSON endpoints as a new Data common feature when we pull out Carto. I can leave it there but at some point we should remove it from the UI until it is rebuilt.

ericyoungberg commented on app/templates/datasets.hbs:
```diff
@@ -54,14 +54,11 @@
             </a>
 
-            {{#if model.raw_data.spatialMetaData}}
+            {{#unless model.raw_data.spatialMetaData}}
```
This will make the spatial button show up only if there isn't spatial metadata

mzagaja replied:
So this particular piece of code is a bit confusing. What it is looking for the spatialMetadata about tabular data, not the spatialMetadata about spatial data. We can refactor it to name this something else, but if you test this on staging or in development you should see that it properly hides the shapefile download for tabular data and shows it for geospatial.

ericyoungberg replied:
```js
model.dataset.get('schemaname')
```

mzagaja committed:
Refactor to Check Schema Name

This refactors the check for displaying the shapefile to check the schema name since there is tabular data that does not include spatial metadata about it.
